p2p/switch.go

```diff
@@ -459,85 +439,73 @@ func (sw *Switch) IsDialingOrExistingAddress(addr *NetAddress) bool {
 		(!sw.config.AllowDuplicateIP && sw.peers.HasIP(addr.IP))
 }
 
-//------------------------------------------------------------------------------------
-// Connection filtering
-
-// FilterConnByAddr returns an error if connecting to the given address is forbidden.
-func (sw *Switch) FilterConnByAddr(addr net.Addr) error {
-	if sw.filterConnByAddr != nil {
-		return sw.filterConnByAddr(addr)
-	}
-	return nil
-}
-
-// FilterConnByID returns an error if connecting to the given peer ID is forbidden.
-func (sw *Switch) FilterConnByID(id ID) error {
-	if sw.filterConnByID != nil {
-		return sw.filterConnByID(id)
-	}
-	return nil
-
-}
-
-// SetAddrFilter sets the function for filtering connections by address.
-func (sw *Switch) SetAddrFilter(f func(net.Addr) error) {
-	sw.filterConnByAddr = f
-}
-
-// SetIDFilter sets the function for filtering connections by peer ID.
-func (sw *Switch) SetIDFilter(f func(ID) error) {
-	sw.filterConnByID = f
-}
+func (sw *Switch) acceptRoutine() {
+	for {
+		p, err := sw.transport.Accept(peerConfig{
+			chDescs:      sw.chDescs,
+			onPeerError:  sw.StopPeerForError,
+			reactorsByCh: sw.reactorsByCh,
+		})
+		if err != nil {
+			switch err.(type) {
+			case *ErrRejected:
+				rErr := err.(*ErrRejected)
```

Comment:
wouldn't it be nicer to have a `IsErrRejected(err error) bool` function

Examples in std lib: os.IsNotExist, os.IsTimeout

```go
// IsTimeout returns a boolean indicating whether the error is known
  // to report that a timeout occurred.
  func IsTimeout(err error) bool {
  	terr, ok := underlyingError(err).(timeout)
  	return ok && terr.Timeout()
  }
  
  // underlyingError returns the underlying error for known os error types.
  func underlyingError(err error) error {
  	switch err := err.(type) {
  	case *PathError:
  		return err.Err
  	case *LinkError:
  		return err.Err
  	case *SyscallError:
  		return err.Err
  	}
  	return err
  }
```

Reply:
We can introduce such a method, yet to get to the methods on the `ErrRejected` you have to assert anyway.